lib/auth/userloginstate/generator.go

```diff
@@ -186,19 +191,19 @@ func (g *Generator) addAccessListsToState(ctx context.Context, user types.User,
 	}
 
 	for _, accessList := range accessLists {
-		if err := services.IsAccessListOwner(identity, accessList); err == nil {
-			g.grantRolesAndTraits(identity, accessList.Spec.OwnerGrants, state)
+		if ownershipType, err := g.memberChecker.IsAccessListOwner(ctx, identity, accessList); err == nil {
+			g.grantRolesAndTraits(identity, accessList.Spec.OwnerGrants, state, ownershipType)
 		}
 
-		if err := g.memberChecker.IsAccessListMember(ctx, identity, accessList); err == nil {
-			g.grantRolesAndTraits(identity, accessList.Spec.Grants, state)
+		if membershipType, err := g.memberChecker.IsAccessListMember(ctx, identity, accessList); err == nil {
+			g.grantRolesAndTraits(identity, accessList.Spec.Grants, state, membershipType)
 		}
 	}
 
 	return nil
 }
 
-func (g *Generator) grantRolesAndTraits(identity tlsca.Identity, grants accesslist.Grants, state *userloginstate.UserLoginState) {
+func (g *Generator) grantRolesAndTraits(identity tlsca.Identity, grants accesslist.Grants, state *userloginstate.UserLoginState, membershipOrOwnershipType services.MembershipOrOwnershipType) accesslist.Grants {
```

Comment:
Grants returned argument seems to be unused: 
```suggestion
func (g *Generator) grantRolesAndTraits(identity tlsca.Identity, grants accesslist.Grants, state *userloginstate.UserLoginState, membershipOrOwnershipType services.MembershipOrOwnershipType {
```